actors_tools/GlobalActorEventPackageReader/AnalysisClassesManager/AnalysisClassesManager.cpp

```diff
@@ -0,0 +1,45 @@
+/**
+ *  @copyright Copyright 2019 The J-PET Gate Authors. All rights reserved.
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  @file AnalysisClassesManager.cpp
+ */
+#include<iostream>
+#include "AnalysisClassesManager.h"
+
+std::unique_ptr<AnalysisClassesManager> AnalysisClassesManager::upInstance;
+
+AnalysisClassesManager::AnalysisClassesManager() { }
+
+AnalysisClassesManager::~AnalysisClassesManager()
+{
+ for ( std::map< std::string, AnalysisClass* >::iterator it = mAnalysisClasses.begin(); it != mAnalysisClasses.end(); ++it ) { delete it->second; }
+ mAnalysisClasses.clear();
+}
+
+AnalysisClassesManager* AnalysisClassesManager::getInstance()
+{
+ if( upInstance.get() == nullptr ) { upInstance.reset( new AnalysisClassesManager() ); }
```

Comment:
`if (upInstance) { ....}`

Reply:
upInstance is unique pointer - so upInstance is an object not a pointer ( method get return pointer to initialized object by unique_pointer ). In this case proposed by you solution is not correct.